OptimizelySDK/Odp/OdpSegmentManager.cs

```diff
@@ -47,13 +47,27 @@ public class OdpSegmentManager : IOdpSegmentManager
         /// </summary>
         private readonly ICache<List<string>> _segmentsCache;
 
-        public OdpSegmentManager(OdpConfig odpConfig, IOdpSegmentApiManager apiManager,
-            int? cacheSize = null, TimeSpan? itemTimeout = null,
-            ILogger logger = null, ICache<List<string>> cache = null
+        public OdpSegmentManager(IOdpSegmentApiManager apiManager,
+            ICache<List<string>> cache = null,
+            ILogger logger = null
+        )
+        {
+            _apiManager = apiManager;
+            _logger = logger ?? new DefaultLogger();
+
+            _segmentsCache =
+                cache ?? new LruCache<List<string>>(Constants.DEFAULT_MAX_CACHE_SIZE,
+                    TimeSpan.FromSeconds(Constants.DEFAULT_CACHE_SECONDS), logger);
+        }
+
+        public OdpSegmentManager(IOdpSegmentApiManager apiManager,
+            int? cacheSize = null,
+            TimeSpan? itemTimeout = null,
+            ICache<List<string>> cache = null,
+            ILogger logger = null
         )
         {
             _apiManager = apiManager;
-            _odpConfig = odpConfig;
             _logger = logger ?? new DefaultLogger();
 
             itemTimeout = itemTimeout ?? TimeSpan.FromSeconds(Constants.DEFAULT_CACHE_SECONDS);
```

Comment:
I see the same negative-timeout filtering and default cache size on nulls are all handled in LRUCache. We can clean up redundant codes from here.